pkg/kv/kvserver/rangefeed/task.go

```diff
@@ -37,48 +37,114 @@ type runnable interface {
 // the Processor was started and hooked up to a stream of logical operations.
 // The Processor can initialize its resolvedTimestamp once the scan completes
 // because it knows it is now tracking all intents in its key range.
+//
+// The task is abstracted so that it can also be used outside the context of a
+// Processor, even though Processor is the primary consumer. This flexibility
+// allows registrations to perform ad-hoc computations of the resolved timestamp
+// at specific Raft log indexes in certain error cases.
```

Comment:
Does `specific Raft log indexes` mean after unsuccessful catch up scan?
We'd rather change original comment to refer to `processorRTSScanConsumer` and give an example of processor being a consumer when running initial scan or ad hoc synchronous consumer run by registration when initial scan can't complete.